src/mbed_tools/build/_internal/config/config.py

```diff
@@ -67,6 +69,20 @@ def _update_config_section(self, config_settings: List[ConfigSetting]) -> None:
 
         self.data[CONFIG_SECTION] = self.data.get(CONFIG_SECTION, []) + config_settings
 
+    def _update_memories_section(self, memories: List[Memory]) -> None:
+        defined_memories = self.data.get(MEMORIES_SECTION, [])
+        for memory in memories:
+            logger.debug(f"Adding memory settings `{memory.name}: start={memory.start} size={memory.size}`")
+            prev_defined = next((mem for mem in defined_memories if mem.name == memory.name), None)
+            if prev_defined is None:
```

Comment:
To clarify, does this code not work for some reason?
```suggestion
        defined_memories = self.data.get(MEMORIES_SECTION, [])
        for memory in memories:
            logger.debug(f"Adding memory settings `{memory.name}: start={memory.start} size={memory.size}`")
            if memory not in defined_memories:
```
Maybe we'll need to tweak the logging message slightly, but using `in` to check for duplicates should be faster and more readable. If I'm missing something and this code won't work then we can go with the original version.

Reply:
The more complicated check is just to prevent multiple definitions for the same memory region, with a different start/size in the config. The `in` check would be faster, but leaves it to the preprocessor to warn about redefining MBED_ROM_START etc. It might not be a frequent enough case to warrant the speed/readability cost of the original version though.

Reply:
Thanks for explaining. Yes, I think the more complex check is warranted if it prevents us running into multiple definition warnings.